File: 백준/Silver/11047. 동전 0/동전 0.py
N, K = tuple(map(int, input().split()))

values = list()
for i in range(N):
    values.append(int(input()))

res = 0
cur_idx = N-1
while(cur_idx >= 0):
    res += int(K/values[cur_idx])
    K = K%values[cur_idx]
    cur_idx -= 1;
    
print(res)

File: 백준/Silver/11047. 동전 0/동전 0.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

readline.on('line', function(line) {
  input.push(line)
}).on('close', function () {
  let [n, k] = input.shift().split(" ").map(Number)

  let answer = 0
  input.reverse()
  for (const coin of input) {
    const cnt = Math.floor(k / coin)
    k -= (cnt * coin)
    answer += cnt
  }

  console.log(answer)
});